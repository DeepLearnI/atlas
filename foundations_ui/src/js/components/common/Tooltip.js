import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Tooltip extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: this.props.message,
    };
  }

  render() {
    const { message } = this.state;
    return (
      <span className="tooltip">{message}</span>
    );
  }
}

Tooltip.propTypes = {
  message: PropTypes.string,
};

Tooltip.defaultProps = {
  message: '',
};

export default Tooltip;
