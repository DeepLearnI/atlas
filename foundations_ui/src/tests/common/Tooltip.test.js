import React from 'react';
import ReactDOM from 'react-dom';
import Tooltip from '../../js/components/common/Tooltip';
import { shallow } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders Tooltip', () => {
  const wrapper = shallow(<Tooltip/>);
  expect(wrapper).toMatchSnapshot();
});