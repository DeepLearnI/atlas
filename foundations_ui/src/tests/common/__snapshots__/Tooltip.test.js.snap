// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Tooltip 1`] = `
<span
  className="tooltip"
/>
`;
